tests/cli/01-help.t

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2018 NIWA
+# 
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# cylc help and basic invocation.
+
+. "$(dirname "$0")/test_header"
+set_test_number 45
+
+# Top help
+run_ok "${TEST_NAME_BASE}-0" cylc
+run_ok "${TEST_NAME_BASE}-h" cylc h
+run_ok "${TEST_NAME_BASE}--h" cylc h
+run_ok "${TEST_NAME_BASE}-help" cylc help
+run_ok "${TEST_NAME_BASE}---help" cylc --help
+cmp_ok "${TEST_NAME_BASE}-0.stdout" "${TEST_NAME_BASE}-h.stdout"
+cmp_ok "${TEST_NAME_BASE}-0.stdout" "${TEST_NAME_BASE}--h.stdout"
+cmp_ok "${TEST_NAME_BASE}-0.stdout" "${TEST_NAME_BASE}-help.stdout"
+cmp_ok "${TEST_NAME_BASE}-0.stdout" "${TEST_NAME_BASE}---help.stdout"
+
+# Sub-command
+run_ok "${TEST_NAME_BASE}-warranty" cylc warranty
+run_ok "${TEST_NAME_BASE}-license-warranty" cylc license warranty
+run_ok "${TEST_NAME_BASE}-GPL-warranty" cylc GPL warranty
+run_ok "${TEST_NAME_BASE}-w" cylc w
+run_ok "${TEST_NAME_BASE}-license-w" cylc license w
+run_ok "${TEST_NAME_BASE}-GPL-w" cylc GPL w
+cmp_ok "${TEST_NAME_BASE}-warranty.stdout" \
+    "${TEST_NAME_BASE}-license-warranty.stdout"
+cmp_ok "${TEST_NAME_BASE}-warranty.stdout" \
+    "${TEST_NAME_BASE}-GPL-warranty.stdout"
+cmp_ok "${TEST_NAME_BASE}-warranty.stdout" "${TEST_NAME_BASE}-w.stdout"
+cmp_ok "${TEST_NAME_BASE}-warranty.stdout" "${TEST_NAME_BASE}-license-w.stdout"
+cmp_ok "${TEST_NAME_BASE}-warranty.stdout" "${TEST_NAME_BASE}-GPL-w.stdout"
+
+# Sub-command - no match
+run_fail "${TEST_NAME_BASE}-aardvark" cylc aardvark
+cmp_ok "${TEST_NAME_BASE}-aardvark.stderr" <<'__STDERR__'
+cylc aardvark: unknown utility. Abort.
+Type "cylc help all" for a list of utilities.
+__STDERR__
+run_fail "${TEST_NAME_BASE}-license-aardvark" cylc license aardvark
+cmp_ok "${TEST_NAME_BASE}-license-aardvark.stderr" <<'__STDERR__'
```

Comment:
Not essential, but could consolidate this comparison with the identical expected output from the above command - indeed this would be better as it would ensure they produce the same.